### PR TITLE
Fix typo

### PR DESCRIPTION
**Pull request by EricCrosson**
Hey there,

Thanks for your work on `nil`! I've been using it for a while now and it is :pinched_fingers: fantastic. I just saw yesterday's release adds go-to definition support for flake inputs, neat! Go-to definition is the biggest feature I miss with Nix LSPs, so I'm happy to see this feature.

Anyway, I saw these minor typos on the readme and figured I'd open a PR.

Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ You can also use this repository as a flake input and add its output to your own
 systemwide or home configuration.
 
 *Disclamer: We ship `flake.lock` which is tested in CI to be working. If you use `follows` to
-override flake inputs, we provides no guarentee of whether it would still build.*
+override flake inputs, we provide no guarentee of whether it would still build.*
```
Comment (notpeelz) [inline README.md:41]: `guarentee` -> `guarantee`

File: README.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
-# nil: NIx Language server
+# nil: Nix Language server
```
Comment (figsoda) [inline README.md:1]: this seems intentional, the capitalized characters spell out nil

Comment (oxalica) [reply README.md:1]: :thinking:  Yeah. But it seems to be confusing for many ones. There's also https://github.com/oxalica/nil/pull/46#discussion_r1051548116
Not sure if there's a better way to express it.

Comment (EricCrosson) [reply README.md:1]: :sweat_smile: I didn't notice that the `s` is lowercase, so the only capitalized letters are `n` `i` `l` 

Egg on my face!

Comment (EricCrosson) [reply README.md]: I removed this diff, but will leave the PR open as it resolves unrelated typos

**Pull request merged**